Beis.LearningPlatform.Web/wwwroot/js/comparisontool.js

```diff
@@ -137,51 +132,31 @@ function validateCompareButton(e) {
 
 function toggleProductsBasedOnCategory() {
     document.querySelector("#allProductsDiv").style.display = "none";
-    this.style.display = "none";
-
-    var selectedProductCategory = document.querySelector("#dummySelectedProductCategory").cloneNode(true);
-    var currentCategoryId = this.getAttribute("data-id");
-    var spanNode = document.createElement("span");
-    spanNode.textContent = this.textContent.trim();
-    selectedProductCategory.appendChild(spanNode);
-    selectedProductCategory.id = "selectedCategoryId-" + currentCategoryId;
-    selectedProductCategory.setAttribute("data-id", currentCategoryId);
-    selectedProductCategory.style = "";
-    selectedProductCategory.style.cursor = "pointer";
-    selectedProductCategory.addEventListener("click", toggleSelectedProductsBasedOnCategory);
-    document.querySelector("#selectedProductCategoryDiv").appendChild(selectedProductCategory);
-
-    var selectedCategoriesCount = document.querySelector("#selectedProductCategoryDiv").childElementCount - 1;
+
     toggleIndividualAndGroupCompareButtons();
     var formDivs = document.querySelectorAll(".formDiv");
-    formDivs.forEach(d => {
-        if (d.id !== "formDiv-" + currentCategoryId && selectedCategoriesCount !== formDivs.length) {
-            d.style.display = "none";
-        } else {
-            d.style.display = "block";
+
+    var allCategoriesCheckBoxes = document.querySelectorAll(".productCategory");
```

Comment:
please use const or let to declare non global variables.

Reply:
Fixed